solr/solrj/src/java/org/apache/solr/client/solrj/request/GenericSolrRequest.java

```diff
@@ -42,6 +44,22 @@ public GenericSolrRequest setContentWriter(ContentWriter contentWriter) {
     return this;
   }
 
+  public GenericSolrRequest withContent(byte[] buf, String type) {
```

Comment:
Isn't it too harmful to be advertised as public for users?

Reply:
I don't think so.
Users can already do so with their own `ContentWriter` impl. This is just a helper method